core/src/modules/configLoaders/destination.ts

```diff
@@ -21,7 +21,11 @@ export async function loadDestination(
 
   const app: App = await getParentByName(App, configObject.appId);
 
-  validateConfigObjectKeys(Destination, configObject);
+  validateConfigObjectKeys(Destination, configObject, [
+    "options",
+    "mapping",
+    "destinationGroupMemberships",
+  ]);
```

Comment:
We now tightly validate the properties from config objects.  Either they are top-level properties we know about (name, type, etc) or this explicit list of extra objects